hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java

```diff
@@ -64,7 +64,9 @@
 import org.apache.hadoop.hbase.MiniHBaseCluster;
 import org.apache.hadoop.hbase.PrivateCellUtil;
 import org.apache.hadoop.hbase.ServerName;
+import org.apache.hadoop.hbase.StartMiniClusterOption;
 import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.TestTableName;
```

Comment:
Cool trick!

But wait, this class is both a `@Rule` and a unit test?!? Nooo!

Reply:
haha ya, noticed that. Weird.

Reply:
https://issues.apache.org/jira/browse/HBASE-23665